```
+ download_release_packages
Traceback (most recent call last):
  File "/usr/bin/download_release_packages", line 90, in <module>
    release = ReleaseInfo(input())
  File "/usr/share/clickhouse-test/ci/get_previous_release_tag.py", line 29, in __init__
    self.type = release_tag[1:].split("-")[1]
IndexError: list index out of range
```

--- a/tests/ci/get_previous_release_tag.py
+++ b/tests/ci/get_previous_release_tag.py
@@ -2,21 +2,25 @@
 
 import re
 import logging
+from typing import List, Optional, Tuple
 
 import requests  # type: ignore
 
 CLICKHOUSE_TAGS_URL = "https://api.github.com/repos/ClickHouse/ClickHouse/tags"
-CLICKHOUSE_PACKAGE_URL = "https://github.com/ClickHouse/ClickHouse/releases/download/v{version}-{type}/clickhouse-common-static_{version}_amd64.deb"
+CLICKHOUSE_PACKAGE_URL = (
+    "https://github.com/ClickHouse/ClickHouse/releases/download/"
+    "v{version}-{type}/clickhouse-common-static_{version}_amd64.deb"
+)
 VERSION_PATTERN = r"(v(?:\d+\.)?(?:\d+\.)?(?:\d+\.)?\d+-[a-zA-Z]*)"
 
 logger = logging.getLogger(__name__)
 
 
 class Version:
-    def __init__(self, version):
+    def __init__(self, version: str):
         self.version = version
 
-    def __lt__(self, other):
+    def __lt__(self, other: "Version") -> bool:
         return list(map(int, self.version.split("."))) < list(
             map(int, other.version.split("."))
         )
@@ -26,7 +30,7 @@ class Version:
 
 
 class ReleaseInfo:
-    def __init__(self, release_tag):
+    def __init__(self, release_tag: str):
         self.version = Version(release_tag[1:].split("-")[0])
         self.type = release_tag[1:].split("-")[1]
 
@@ -37,7 +41,9 @@ class ReleaseInfo:
         return f"ReleaseInfo: {self.version}-{self.type}"
 
 
-def find_previous_release(server_version, releases):
+def find_previous_release(
+    server_version: Optional[Version], releases: List[ReleaseInfo]
+) -> Tuple[bool, Optional[ReleaseInfo]]:
     releases.sort(key=lambda x: x.version, reverse=True)
 
     if server_version is None:
@@ -59,21 +65,23 @@ def find_previous_release(server_version, releases):
                 != 404
             ):
                 return True, release
-            else:
-                logger.debug(
-                    "The tag %s-%s exists but the package is not yet available on GitHub",
-                    release.version,
-                    release.type,
-                )
+
+            logger.debug(
+                "The tag v%s-%s exists but the package is not yet available on GitHub",
+                release.version,
+                release.type,
+            )
 
     return False, None
 
 
-def get_previous_release(server_version):
+def get_previous_release(server_version: Optional[Version]) -> Optional[ReleaseInfo]:
     page = 1
     found = False
     while not found:
-        response = requests.get(CLICKHOUSE_TAGS_URL, {"page": page, "per_page": 100})
+        response = requests.get(
+            CLICKHOUSE_TAGS_URL, {"page": page, "per_page": 100}, timeout=10
+        )
         if not response.ok:
             raise Exception(
                 "Cannot load the list of tags from github: " + response.reason
@@ -94,7 +102,11 @@ def get_previous_release(server_version):
     return previous_release
 
 
-if __name__ == "__main__":
+def main():
     logging.basicConfig(level=logging.INFO)
     server_version = Version(input())
     print(get_previous_release(server_version))
+
+
+if __name__ == "__main__":
+    main()

--- a/tests/ci/get_previous_release_tag.py
+++ b/tests/ci/get_previous_release_tag.py
@@ -54,10 +54,7 @@ def find_previous_release(server_version, releases):
                     CLICKHOUSE_PACKAGE_URL.format(
                         version=release.version, type=release.type
                     ),
-                    total=10,
-                    read=10,
-                    connect=10,
-                    backoff_factor=0.3,
+                    timeout=10,
                 ).status_code
                 != 404
             ):

--- a/tests/ci/get_previous_release_tag.py
+++ b/tests/ci/get_previous_release_tag.py
@@ -9,6 +9,8 @@ CLICKHOUSE_TAGS_URL = "https://api.github.com/repos/ClickHouse/ClickHouse/tags"
 CLICKHOUSE_PACKAGE_URL = "https://github.com/ClickHouse/ClickHouse/releases/download/v{version}-{type}/clickhouse-common-static_{version}_amd64.deb"
 VERSION_PATTERN = r"(v(?:\d+\.)?(?:\d+\.)?(?:\d+\.)?\d+-[a-zA-Z]*)"
 
+logger = logging.getLogger(__name__)
+
 
 class Version:
     def __init__(self, version):
@@ -61,10 +63,10 @@ def find_previous_release(server_version, releases):
             ):
                 return True, release
             else:
-                print(
-                    "The tag {version}-{type} exists but the package is not yet available on GitHub".format(
-                        version=release.version, type=release.type
-                    )
+                logger.debug(
+                    "The tag %s-%s exists but the package is not yet available on GitHub",
+                    release.version,
+                    release.type,
                 )
 
     return False, None

